apps/pollbook/barcode-scanner-daemon/src/unitech_ts100_daemon.rs

```diff
@@ -0,0 +1,270 @@
+        if desc.vendor_id() == 0x2745 && desc.product_id() == 0x300a {
+            let mut handle: DeviceHandle<_> = device.open()?;
+            handle.reset()?;
+            sleep(std::time::Duration::from_millis(500));
+            break;
+        }
+    }
+    Ok(())
+}
+
+fn init_port(
+    port_name: &str,
+    baud_rate: u32,
+) -> color_eyre::Result<Box<dyn serialport::SerialPort>> {
+    // We have experienced difficulty reconnecting the scanner when the daemon
+    // is stopped and started multiple times. Resetting the scanner solves the issue.
+    match reset_scanner() {
+        Ok(_) => {
+            log!(
+                event_id: EventId::UsbDeviceReconnectAttempted,
+                message: "Barcode scanner reset succeeded".to_string(),
+                event_type: EventType::SystemAction,
+                disposition: Disposition::Success
+            )
+        }
+        Err(e) => {
+            log!(
+                event_id: EventId::UsbDeviceReconnectAttempted,
+                message: format!("Barcode scanner reset failed: {e}"),
+                event_type: EventType::SystemAction,
+                disposition: Disposition::Failure
+            );
+
+            log!(
+                EventId::ProcessTerminated;
+                EventType::SystemAction
+            );
+            exit(1);
+        }
+    }
+
+    // Open barcode scanner device
+    let port = serialport::new(port_name, baud_rate)
+        .data_bits(DataBits::Eight)
+        .parity(Parity::None)
+        .stop_bits(StopBits::One)
+        .flow_control(FlowControl::None)
+        .timeout(Duration::from_millis(500))
+        .open()
+        .with_context(|| format!("Failed to open serial port {}", port_name))?;
+
+    Ok(port)
+}
+
+fn main() -> color_eyre::Result<()> {
+    color_eyre::install()?;
+
+    set_source(SOURCE);
+    log!(
+        EventId::ProcessStarted;
+        EventType::SystemAction
+    );
+
+    let _ = fs::remove_file(UDS_PATH);
+    let listener = UnixListener::bind(UDS_PATH)?;
+
+    // TODO(Kevin) run server and daemon as same user so we can avoid lax permissions
+    fs::set_permissions(UDS_PATH, fs::Permissions::from_mode(0o666))?;
+
+    // A thread‐safe list of all currently‐connected clients
+    let clients = Arc::new(Mutex::new(Vec::<UnixStream>::new()));
+
+    // Spawn thread to accept clients
+    {
+        let clients = clients.clone();
+        thread::spawn(move || {
+            for stream in listener.incoming() {
+                match stream {
+                    Ok(sock) => {
+                        clients.lock().unwrap().push(sock);
+                    }
+                    Err(e) => {
+                        log!(event_id: EventId::SocketConnection, message: format!("UDS client manager encountered error with client: {}", e), event_type: EventType::SystemStatus, disposition: Disposition::Failure)
+                    }
+                }
+            }
+        });
+    }
```

Comment:
We expect exactly one client so this might be overkill. Soon we'll add UDS reconnection logic to the pollbook backend. One could look at this as resiliency in case we fail to clean up UDS clients created during reconnection. But probably better to just be thorough in our clean up